### PR TITLE
Swift 5

### PR DESCRIPTION
**Pull request by stephencelis**
Swift 5 snapshot testing!

### CODE REVIEW

File: Sources/SnapshotTesting/SnapshotTestCase.swift [hunk 1]
```diff
@@ -1,256 +0,0 @@
-import XCTest
```
Comment (stephencelis) [inline Sources/SnapshotTesting/SnapshotTestCase.swift]: 👋

**Comment by stephencelis** [comment]
I think the macOS snaps are failing for dark/light/mojave/beta reasons.

**Comment by mbrandonw** [comment]
> I think the macOS snaps are failing for dark/light/mojave/beta reasons.

In the circle logs it seems to be a crash:

> Restarting after unexpected exit or crash in SnapshotTestingTests/testMixedViews(); summary will include totals from previous launches.

File: Makefile [hunk 2]
```diff
@@ -15,7 +15,7 @@ test-ios:
 	set -o pipefail && \
 	xcodebuild test \
 		-scheme SnapshotTesting_iOS \
-		-destination platform="iOS Simulator,name=iPhone XR,OS=12.1" \
+		-destination platform="iOS Simulator,name=iPhone Xʀ,OS=12.2" \
```
Comment (mbrandonw) [inline Makefile]: smh

File: Sources/SnapshotTesting/AssertSnapshot.swift [hunk 3]
```diff
@@ -65,9 +64,9 @@ public func assertSnapshots<Value, Format>(
   line: UInt = #line
   ) {
 
-  strategies.forEach { name, strategy in
+  try? strategies.forEach { name, strategy in
```
Comment (mbrandonw) [inline Sources/SnapshotTesting/AssertSnapshot.swift]: should we make `assertSnapshots` throw? on this cause any `value: () -> Value` throws to be swallowed?

File: Sources/SnapshotTesting/SnapshotTestCase.swift [hunk 4]
```diff
@@ -1,256 +1,4 @@
 import XCTest
 
-#if !os(Linux)
 @available(swift, obsoleted: 5.0, renamed: "XCTestCase", message: "Please use XCTestCase instead")
 public typealias SnapshotTestCase = XCTestCase
-#else
-
-/// An XCTest subclass that provides snaphot testing helpers.
-open class SnapshotTestCase: XCTestCase {
```
Comment (mbrandonw) [inline Sources/SnapshotTesting/SnapshotTestCase.swift]: 👋

**Comment by mbrandonw** [comment]
are you sure you want to record the precision test?

image

I think we want to test that they match even though they are slightly different.

Wonder if we should start checking if the snapshot "matches" before recording, but that would definitely increase record times by quite a bit :/

**Pull request merged**